app/assets/javascripts/discourse/app/services/document-title.js

```diff
@@ -0,0 +1,105 @@
+import Service from "@ember/service";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+
+export default Service.extend({
+  appEvents: service(),
+  contextCount: null,
+  notificationCount: null,
+  _title: null,
+  _backgroundNotify: null,
+
+  init() {
+    this._super(...arguments);
+    this.reset();
+  },
+
+  reset() {
+    this.contextCount = 0;
+    this.notificationCount = 0;
+    this._title = null;
+    this._backgroundNotify = null;
+  },
+
+  getTitle() {
+    return this._title;
+  },
+
+  setTitle(title) {
+    this._title = title;
+    this._renderTitle();
+  },
```

Comment:
`getTitle()`? `setTitle()`? What is this, java? 😜

That looks like a native getters/setters opportunity. 😉 

```js
  get title() {
    return this._title;
  },

  set title(title) {
    this._title = title;
    this._renderTitle();
  },
```

(Possibly the same with with `updateFocus` and `updateContextCount` below)

Reply:
Agreed, I feel like we will have to draw some line in the next months, we don't have any history/patterns on this so far

Reply:
There is one! (since a year ago!) https://github.com/discourse/discourse/blame/8a9e4504fea9edd693ceecf314308cd08e99f0ca/app/assets/javascripts/discourse/app/services/emoji-store.js#L20-L34 😅

Reply:
I think I'll leave the set* methods for now (updated focus to use the same naming) and we can revisit with setters later.